UI revamp changes

satyambnsal opened the pull request:
Changes summary

#147 
#148 
#149 
#150 
#151 
#114 
#38

satyambnsal committed:
Add Tab structure

satyambnsal committed:
Plugin info tab changes

satyambnsal committed:
Fix tab padding issue

satyambnsal committed:
Fix button colors and text scheme

satyambnsal committed:
Transaction card  changes

satyambnsal committed:
Plugin footer changes

satyambnsal committed:
Settings tab changes

satyambnsal committed:
Compiler version changes

satyambnsal committed:
Indentation formatting changes

varex83 commented:
![image](https://github.com/user-attachments/assets/ed3e1832-24e0-4a46-a44b-4adf43914830)

I have some problems with paddings/margin here, can you fix it to be centered?

varex83 commented:
![image](https://github.com/user-attachments/assets/75bfd332-fd05-4d39-a106-7f92641d7176)

also, remove this shadow under the env section

varex83 commented:
![image](https://github.com/user-attachments/assets/14ae7ed7-6bbe-4694-8292-68d271737dc0)

nitpick, make it more table-like, and use tx hash instead of txid

varex83 commented:
![image](https://github.com/user-attachments/assets/5365450a-6490-4934-a5ed-6bff7fe57705)

can we have the same width of dropdown as selector?

varex83 commented on plugin/src/components/Settings/index.tsx:
```diff
@@ -0,0 +1,96 @@
+  const [solidityVersion, setSolidityVersion] = useAtom(solidityVersionAtom)
+  const [versions, setVersions] = useAtom(versionsAtom)
+
+  const [, refetchVersions] = useAsyncFn(async () => {
```
with this approach, we are loading versions only when rendering "Settings" tab, so each time we need to go here and wait until plugin fetch it, but it should done as soon as plugin loads

satyambnsal replied:
Ah good catch!!. Thank you. I moved it to CompilerVersion component so it loads on initial render.

varex83 commented on plugin/src/features/CompilerVersion/index.tsx:
```diff
@@ -1,26 +1,16 @@
-import React, { useEffect } from 'react'
```
Maybe we should rename `CompilerVersion` to something like `Header` or smth

satyambnsal committed:
Move solidity allowed version fetch to compiler version component

satyambnsal committed:
Rename compiler version to header

satyambnsal committed:
Fix margin padding issue for compiler accordian

satyambnsal committed:
Fix Dropdown item width issue

satyambnsal committed:
Change TxId to TxnHash

satyambnsal commented:
@varex83 Thank you for the thorough review. I've addressed all the points raised, with one exception: changing the Transaction history table structure to a more table-like format. This modification would require good CSS changes, so I propose we handle this in a separate PR.
Here is how it looks after changing TxId to TxHash

<img width="490" alt="Screenshot 2024-07-26 at 3 50 46 PM" src="https://github.com/user-attachments/assets/ee56b582-4238-41c5-847f-7576b5377fb9">

satyambnsal committed:
Accordian shadow and button margin css changes

satyambnsal commented:
Review changes done.

varex83 reviewed (changes requested):
![image](https://github.com/user-attachments/assets/facdc72a-2400-4df9-b6d0-c423471bb323)

can you quickly add some margings there, and we are good to go?

satyambnsal committed:
Add gap in Environment text

varex83 reviewed (approved):
LGTM
